plugins/org.eclipse.sirius.tests.swtbot/src/org/eclipse/sirius/tests/swtbot/LinkedBorderNodeLocationTest.java

```diff
@@ -0,0 +1,167 @@
+        checkBorderNodePosition(getSubContainerBounds(SUB_CONTAINER), getBorderNodeBounds(BORDER_NODE_2), PositionConstants.EAST);
+        checkBorderNodePosition(getSubContainerBounds(SUB_CONTAINER), getBorderNodeBounds(BORDER_NODE_3), PositionConstants.SOUTH);
+        checkBorderNodePosition(getSubContainerBounds(SUB_CONTAINER), getBorderNodeBounds(BORDER_NODE_4), PositionConstants.SOUTH);
+    }
+
+    private void dragAndDropSubContainer(String targetName) {
+        Point subEPackage1Center = getSubContainerBounds(SUB_CONTAINER).getCenter().getCopy();
+        Point targetCenter = getContainerBounds(targetName).getCenter().getCopy();
+        CheckEditPartMoved movedCondition = new CheckEditPartMoved(editor, SUB_CONTAINER, DNodeContainer2EditPart.class, getSubContainerBounds(SUB_CONTAINER).getLocation());
+        editor.drag(subEPackage1Center, targetCenter);
+        bot.waitUntil(movedCondition);
+    }
+
+    private Rectangle getBorderNodeBounds(String borderNodeName) {
+        return editor.getBounds(editor.getEditPart(borderNodeName, AbstractDiagramBorderNodeEditPart.class));
+    }
+
+    private Rectangle getSubContainerBounds(String containerName) {
+        return editor.getBounds(editor.getEditPart(containerName, DNodeContainer2EditPart.class));
+    }
+
+    private Rectangle getContainerBounds(String containerName) {
+        return editor.getBounds(editor.getEditPart(containerName, DNodeContainerEditPart.class));
+    }
+
+    /**
+     * Checks that a border node is on the correct side of its bordered node. Magic numbers are used here, corresponding
+     * to the delta between the border of the bordered node and the closest border of the border node.
+     * 
+     * @param borderedNodeBounds
+     *            the bordered node rectangle
+     * @param borderNodeBounds
+     *            the border node rectangle
+     * @param expectedSide
+     *            the side where the border node is expected among {@code PositionConstants.NORTH},
+     *            {@code PositionConstants.EAST}, {@code PositionConstants.SOUTH}, {@code PositionConstants.WEST}.
+     */
+    private static void checkBorderNodePosition(Rectangle borderedNodeBounds, Rectangle borderNodeBounds, int expectedSide) {
+        switch (expectedSide) {
+        case PositionConstants.NORTH:
+            assertEquals(borderedNodeBounds.getTop().y, borderNodeBounds.getBottom().y - 8, 2);
+            break;
+        case PositionConstants.EAST:
+            assertEquals(borderedNodeBounds.getRight().x, borderNodeBounds.getLeft().x + 10, 2);
+            break;
+        case PositionConstants.SOUTH:
+            assertEquals(borderedNodeBounds.getBottom().y, borderNodeBounds.getTop().y + 10, 2);
+            break;
+        case PositionConstants.WEST:
+            assertEquals(borderedNodeBounds.getLeft().x, borderNodeBounds.getRight().x - 8, 2);
+            break;
+        }
+
+    }
```

Comment:
Note: I'm just wondering if magic numbers are reliable here, have you looked at other test examples like class ArrangeAllLinkedBorderedNodesLayoutStabilityTest?

Reply:
I looked at this class, but couldn't find a method corresponding to what I wanted to do; this method seems clearer to me
I see that the `org.eclipse.sirius.tests.swtbot.ArrangeAllLinkedBorderedNodesLayoutStabilityTest.validateEdgeFromPortCrossParentContainer(AbstractBorderedShapeEditPart, AbstractDiagramBorderNodeEditPart)` method already uses a safety margin of 5 pixels, and the idea is similar here with a delta of 2 pixels. The other numbers correspond to the offset between the border node and its container.
I'll turn them into constants to make them more meaningful